Middleware/checkAuthStatus.js

```diff
@@ -0,0 +1,33 @@
+import dotenv from 'dotenv';
+
+import jwt from 'jsonwebtoken';
+import User from '../Models/users.js';
+
+dotenv.config();
+
+const checkAuthStatus = async (req, res, next) => {
+  const token = req.headers.authorization?.split(' ')[1];
+
+  if (!token) {
+    return res
+      .status(401)
+      .json({ error: 'You must be logged in to make this request!' });
+  }
+
+  try {
+    const { id } = jwt.verify(
+      token,
+      process.env.MY_JWT_SECRET
+        || 'S2T7iqfnSIL1RWP9N8BCCs5jEgDwYRJ0ZbzNA6XF43dO',
+    );
```

Comment:
I'm wondering, what if the verfiy method does not return a valid id?
I mean, yes, you are providing an actual valid JWT secret here, but if you remove that secret at some point you could receive a token which is or has become invalid.

Reply:
When I generate my token, I use the **id** and the **secret**. 
The verify method uses the **secret** to get the **id** back.

The secret should always stay the same. In production it‘s **hardcoded** into the system.

Or did I not understand your question? 😄

Reply:
You did understand my question 😄 I did not understand that the secret never changes. If that's the case, me happy 🥳